smithy-linters/src/main/java/software/amazon/smithy/linters/CamelCaseValidator.java

```diff
@@ -135,14 +157,66 @@ public List<ValidationEvent> validate(Model model) {
                         shape.getType(), shape.getId().getName())))
                 .forEach(events::add);
 
-        Pattern isValidMemberName = config.getMemberNames().getRegex();
-        model.shapes(MemberShape.class)
-                .filter(shape -> !isValidMemberName.matcher(shape.getMemberName()).find())
-                .map(shape -> danger(shape, format(
-                        "Member shape member name, `%s`, is not %s camel case",
-                        shape.getMemberName(), config.getMemberNames())))
-                .forEach(events::add);
+        // First validate all member shapes connected to services
```

Comment:
This is close, but you should group by service closures not validate that any shape in any service closure uses consistent casing. For example, a model might contain two services: one service uses lower case members, and the other uses upper case. We would want to emit correctly for both of these.